### PR TITLE
feature/signup to develop branch

### PR DESCRIPTION
**Pull request by Umaid-gh**
More changes are required. Initial draft created.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -1,75 +1,73 @@
...
-	<description>New project for AIYKR</description>
-	<properties>
-		<java.version>11</java.version>
-	</properties>
-	<dependencies>
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-data-mongodb</artifactId>
-		</dependency>
-		<!-- Spring Boot Starter Data JPA includes Hibernate -->
```
Comment (Faizan-Ahmed-Khan) [inline pom.xml:24]: Remove if not required

File: pom.xml [hunk 2]
```diff
@@ -1,75 +1,73 @@
...
-<!--		</dependency>-->
-		<dependency>
-			<groupId>org.springframework.boot</groupId>
-			<artifactId>spring-boot-starter-web</artifactId>
-		</dependency>
-<!--		<dependency>-->
-<!--			<groupId>org.springframework.boot</groupId>-->
-<!--			<artifactId>spring-boot-devtools</artifactId>-->
-<!--			<scope>runtime</scope>-->
-<!--			<optional>true</optional>-->
```
Comment (Faizan-Ahmed-Khan) [inline pom.xml:38]: Remove if not required

File: src/main/java/com/aiykr/iquais/controller/UserController.java [hunk 3]
```diff
@@ -24,7 +27,7 @@ public ResponseEntity<Response<List<UserResponseDTO>>> getAllUsers() {
 		return userService.getAllUsers();
 	}
 	@PostMapping("/users/student")
-	public ResponseEntity<Response> createStudent(@RequestBody PostUserDTO postUserDTO) throws ResponseExceptionHandler {
+	public ResponseEntity<Response> createStudent(@RequestBody PostUserDTO postUserDTO) throws IquaisException {
 		return userService.createUser(postUserDTO);
```
Comment (Faizan-Ahmed-Khan) [inline src/main/java/com/aiykr/iquais/controller/UserController.java]: Should the service be directly creating a Response Entity? shouldn't it be the controllers job

File: src/main/java/com/aiykr/iquais/repository/UserRepository.java [hunk 4]
```diff
@@ -5,7 +5,10 @@
 import org.springframework.data.mongodb.repository.MongoRepository;
 import org.springframework.stereotype.Repository;
 
+import java.util.List;
+
 @Repository
 public interface UserRepository extends MongoRepository<UserDAO, ObjectId> {
```
Comment (Faizan-Ahmed-Khan) [inline src/main/java/com/aiykr/iquais/repository/UserRepository.java]: Method level comment missing

File: pom.xml [hunk 5]
```diff
@@ -1,75 +1,73 @@
...
+            <artifactId>modelmapper</artifactId>
+            <version>3.1.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.32</version> <!-- Use the latest version -->
+        </dependency>
+        <dependency>
+            <groupId>ch.qos.logback</groupId>
```
Comment (Faizan-Ahmed-Khan) [inline pom.xml]: whats the need for this?

File: pom.xml [hunk 6]
```diff
@@ -1,75 +1,73 @@
...
+        <!-- https://mvnrepository.com/artifact/org.modelmapper/modelmapper -->
+        <dependency>
+            <groupId>org.modelmapper</groupId>
+            <artifactId>modelmapper</artifactId>
+            <version>3.1.1</version>
+        </dependency>
+        <dependency>
+            <groupId>org.slf4j</groupId>
+            <artifactId>slf4j-api</artifactId>
+            <version>1.7.32</version> <!-- Use the latest version -->
```
Comment (Faizan-Ahmed-Khan) [inline pom.xml]: Is this required

File: src/main/java/com/aiykr/iquais/service/UserService.java [hunk 7]
```diff
@@ -3,13 +3,13 @@
...
 import com.aiykr.iquais.dto.response.UserResponseDTO;
-import com.aiykr.iquais.exception.ResponseExceptionHandler;
+import com.aiykr.iquais.exception.IquaisException;
 import org.springframework.http.ResponseEntity;
 
 import java.util.List;
 
 public interface UserService {
-    public ResponseEntity<Response> createUser(PostUserDTO postUserDTO) throws ResponseExceptionHandler;
+    public ResponseEntity<Response> createUser(PostUserDTO postUserDTO) throws IquaisException;
```
Comment (Faizan-Ahmed-Khan) [inline src/main/java/com/aiykr/iquais/service/UserService.java]: Arent methods declared in interface by default public?

File: src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java [hunk 8]
```diff
@@ -23,40 +25,67 @@
...
+                guardianDAO.setType(UserType.GUARDIAN.name());
+                userRepository.save(guardianDAO);
+                log.info("Save New Guardian: {}", guardianDAO);
+            }
+        } catch (IquaisException iqEx) {
+            throw iqEx;
+        }
+        catch (Exception ex) {
+            log.error("An exception occurred: {}", ex.getMessage(), ex);
+            ex.getStackTrace();
```
Comment (Faizan-Ahmed-Khan) [inline src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java]: remove this

File: pom.xml (hunk 1 above)
Comment (Umaid-gh) [reply pom.xml:24]: not clear

File: pom.xml (hunk 2 above)
Comment (Umaid-gh) [reply pom.xml:38]: removed commented lines

File: src/main/java/com/aiykr/iquais/repository/UserRepository.java (hunk 4 above)
Comment (Umaid-gh) [reply src/main/java/com/aiykr/iquais/repository/UserRepository.java]: added in the code now

File: src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java (hunk 8 above)
Comment (Umaid-gh) [reply src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java]: did understand as the line is not mentioned clearly

File: pom.xml (hunk 5 above)
Comment (Umaid-gh) [reply pom.xml]: Including Logback in your project allows you to use its logging capabilities, such as logging messages at different levels (info, debug, error, etc.)

File: pom.xml (hunk 6 above)
Comment (Umaid-gh) [reply pom.xml]: version not required here

File: src/main/java/com/aiykr/iquais/service/UserService.java (hunk 7 above)
Comment (Umaid-gh) [reply src/main/java/com/aiykr/iquais/service/UserService.java]: removed

File: src/main/java/com/aiykr/iquais/controller/UserController.java (hunk 3 above)
Comment (Umaid-gh) [reply src/main/java/com/aiykr/iquais/controller/UserController.java]: changed as per suggestion

File: src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java [hunk 9]
```diff
@@ -33,10 +34,8 @@ public class UserServiceImpl implements UserService {
 
     private static final Logger log = LoggerFactory.getLogger(UserServiceImpl.class);
 
-    private static final String ERROR_CODE_IQ001 = "IQ001";
-
     @Autowired
-    UserRepository userRepository;
+    private UserRepository userRepository;
```
Comment (Faizan-Ahmed-Khan) [inline src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java]: rename Interface to IUserRepository

File: src/main/java/com/aiykr/iquais/dto/response/ErrorCodes.java [hunk 10]
```diff
@@ -0,0 +1,14 @@
...
+
+public class ErrorCodes {
+    public static final String IQ001 = "IQ001";
+    public static final String IQ002 = "IQ002";
+    public static final String IQ003 = "IQ003";
+    public static final String IQ005 = "IQ005";
+    // Add more error codes as needed
+
+    // Private constructor to hide the implicit public one
+    private ErrorCodes() {
```
Comment (Faizan-Ahmed-Khan) [inline src/main/java/com/aiykr/iquais/dto/response/ErrorCodes.java:35]: Do we need to Have this constructor?

Comment (Umaid-gh) [reply src/main/java/com/aiykr/iquais/dto/response/ErrorCodes.java:35]: it suggested in sonarlint

File: src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java (hunk 9 above)
Comment (Umaid-gh) [reply src/main/java/com/aiykr/iquais/service/impl/UserServiceImpl.java]: done

**Review by Faizan-Ahmed-Khan — APPROVED** [review]
LGTM

**Pull request merged**